### PR TITLE
Add nf-core bot to automate changelog, fix linting, update snapshots

### PR DESCRIPTION
**Pull request by jonasscheid**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mhcquant/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mhcquant _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: .github/workflows/changelog.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          pip install pyyaml
+
+      - name: Update CHANGELOG.md from the PR title
+        env:
+          COMMENT: ${{ github.event.comment.body }}
+          GH_TOKEN: ${{ secrets.NF_CORE_BOT_AUTH_TOKEN }}
+          PR_NUMBER: ${{ github.event.issue.number || github.event.pull_request.number }}
+          PR_TITLE: ${{ github.event.issue.title || github.event.pull_request.title }}
+        run: |
+          python ${GITHUB_WORKSPACE}/.github/workflows/changelog.py
```
Comment (mashehu) [inline .github/workflows/changelog.yml:55]: you need to also add this script to make this action work

File: .github/workflows/update-snapshots.yml
Comment (mashehu) [inline .github/workflows/update-snapshots.yml:1]: these are the wrong snapshots, this tests our TUI.
You are looking for something like https://github.com/nf-core/modules/blob/master/.github/workflows/update-gpu-snapshot.yml

Comment (jonasscheid) [reply .github/workflows/update-snapshots.yml:1]: Thanks for the pointer! I tried to implement it according to this workflow. Is there any way to dry-run it?

Comment (jonasscheid) [reply .github/workflows/update-snapshots.yml:1]: @mashehu Is that what you had in mind?

**Comment by FriederikeHanssen** [comment]
would it make sense to add this to the template?

**Comment by mashehu** [comment]
Maybe, but we usually go the: first have it in some pipelines to find edge cases and then add it to the template

**Comment by jonasscheid** [comment]
I volunteer 👋🏼 Let's merge this one soon, then I can test

File: .github/workflows/update-snapshots.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
+name: Update Textual snapshots from a comment
```
Comment (mirpedrol) [inline .github/workflows/update-snapshots.yml]:
```suggestion
name: Update nf-test snapshots from a comment
```

**Review by mirpedrol — APPROVED** [review]
